tasks/ThrusterLimitationTask.cpp

```diff
@@ -0,0 +1,104 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "ThrusterLimitationTask.hpp"
+#include <algorithm>
+
+using namespace std;
+using namespace base;
+using namespace gazebo_usv;
+using namespace control_base;
+
+ThrusterLimitationTask::ThrusterLimitationTask(std::string const& name)
+    : ThrusterLimitationTaskBase(name)
+{
+}
+
+ThrusterLimitationTask::~ThrusterLimitationTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See ThrusterLimitationTask.hpp for more detailed
+// documentation about them.
+
+bool ThrusterLimitationTask::configureHook()
+{
+    if (!ThrusterLimitationTaskBase::configureHook()) {
+        return false;
+    }
+    m_limits = _limits.get();
+    if (m_limits.elements.empty()) {
+        // Initialize speed limits as infinity
+        JointLimits infinity;
+        JointLimitRange range;
+        range.Speed(-base::infinity<float>(), base::infinity<float>());
+        infinity.elements.push_back(range);
+        m_limits = infinity;
+    }
+
+    return true;
+}
+
+bool ThrusterLimitationTask::startHook()
+{
+    if (!ThrusterLimitationTaskBase::startHook()) {
+        return false;
+    }
+    return true;
+}
+
+bool ThrusterLimitationTask::checkSpeedSaturation(commands::Joints const& cmd)
+{
+    return cmd.elements[0].speed >= m_limits.elements[0].max.speed ||
+           cmd.elements[0].speed <= m_limits.elements[0].min.speed;
+}
+
+void ThrusterLimitationTask::evaluateSpeedCommand(commands::Joints const& cmd)
```

Comment:
```suggestion
void ThrusterLimitationTask::validateSpeedCommand(commands::Joints const& cmd)
```

This is more of a validation than an evaluation